flang/lib/Semantics/check-allocate.cpp

```diff
@@ -495,23 +498,24 @@ bool AllocationCheckerHelper::RunChecks(SemanticsContext &context) {
           "Allocatable object in ALLOCATE must be type compatible with type-spec"_err_en_US);
       return false;
     }
+    if (!HaveSameAssumedTypeParameters(*type_, *allocateInfo_.typeSpec)) {
+      // C935
```

Comment:
Use the new constraint number from the latest standard ("F'2023 C939").

Reply:
Ack